src/modules/dashboard/components/PermissionManagementDialog/PermissionManagementCheckbox.tsx

```diff
@@ -0,0 +1,146 @@
+      'dashboard.PermissionManagementDialog.PermissionManagementCheckbox.roleDescription1',
+    defaultMessage: 'Take actions effecting the colony as a whole.',
+  },
+  roleDescription2: {
+    id:
+      // eslint-disable-next-line max-len
+      'dashboard.PermissionManagementDialog.PermissionManagementCheckbox.roleDescription2',
+    defaultMessage: 'Coming soon...',
+  },
+  roleDescription3: {
+    id:
+      // eslint-disable-next-line max-len
+      'dashboard.PermissionManagementDialog.PermissionManagementCheckbox.roleDescription3',
+    defaultMessage: 'Set permissions in the active domain, and any subdomain.',
+  },
+  // We don't have architecture_subdomain (which would be 4)
+  roleDescription5: {
+    id:
+      // eslint-disable-next-line max-len
+      'dashboard.PermissionManagementDialog.PermissionManagementCheckbox.roleDescription5',
+    defaultMessage: 'Fund expenditures and transfer funds between domains.',
+  },
+  roleDescription6: {
+    id:
+      // eslint-disable-next-line max-len
+      'dashboard.PermissionManagementDialog.PermissionManagementCheckbox.roleDescription6',
+    defaultMessage: 'Create and manage expenditures.',
+  },
+  tooltipNoPermissionsText: {
+    id:
+      // eslint-disable-next-line max-len
+      'dashboard.PermissionManagementDialog.PermissionManagementCheckbox.tooltipNoPermissionsText',
+    defaultMessage: 'You do not have permission to set the {roleName} role.',
+  },
+  tooltipNoRootDomainSelected: {
+    id:
+      // eslint-disable-next-line max-len
+      'dashboard.PermissionManagementDialog.PermissionManagementCheckbox.tooltipNoRootDomainSelected',
+    defaultMessage: 'Switch domain to #Root to set the root role.',
+  },
+});
+
+interface Props {
+  asterisk: boolean;
+  disabled: boolean;
+  role: ColonyRole;
+  domainId: number;
+}
+
+const displayName =
+  'dashboard.PermissionManagementDialog.PermissionManagementCheckbox';
+
+const PermissionManagementCheckbox = ({
+  asterisk,
+  disabled,
+  role,
+  domainId,
+}: Props) => {
+  const roleNameMessage = { id: `role.${role}` };
+  const roleDescriptionMessage = useMemo(
+    () =>
+      MSG[`roleDescription${role}`] || {
+        id: '',
+        defaultMessage: '',
+      },
+    [role],
+  );
+
+  const { formatMessage } = useIntl();
+
+  const formattedRole = formatMessage(roleNameMessage);
+
+  const tooltipText =
+    domainId !== ROOT_DOMAIN_ID && formattedRole === 'Root'
+      ? MSG.tooltipNoRootDomainSelected
+      : MSG.tooltipNoPermissionsText;
+
+  const formattedTooltipText = formatMessage(tooltipText, {
+    roleName: formattedRole.toLowerCase(),
+  });
+
+  return (
+    <Checkbox
+      className={styles.permissionChoice}
+      value={role}
+      name="roles"
+      disabled={disabled}
+      tooltipText={formattedTooltipText}
+      tooltipPopperProps={{
+        modifiers: [
+          {
+            name: 'offset',
+            options: {
+              offset: [130, 10],
```

Comment:
The problem is that there's a strange behaviour within the Tooltip. If you don't disabled `Prevent Overflow` on the popper then the `skid` value is not going to work, and I think it's because that modifier is putting in his own `skid` values to wrongly prevent the popper from overflowing the Dialog.

What I did at first was disable the `Prevent Overflow`, but as you saw, the `skid` value is incremented for each iteration.

So the solution to this is what I recently uploaded. Instead of relying on the `skid` value, we rely on the padding to center the popper, and with that, it's also not necessary the disablement of `Prevent Overflow.`

Reply:
I'll have a look at how it renders locally. If it's all good we'll go with this solution for now.

Reply:
![Screenshot from 2020-12-03 18-05-46](https://user-images.githubusercontent.com/1193222/101055015-3247dd80-3592-11eb-82ae-797f1cb3a9eb.png)

This works fine now. Good job! :100: